content/docs/learning-guide/tutorials/rendering/vegetation-bending-tutorial.md

```diff
@@ -0,0 +1,857 @@
+
+Amazing, your tree now sways and reacts to wind! Try to place multiple trees and observe how the trees sway differently when close together versus farther away. Also, add some lighting to make the trees pop!
+
+{{< video src="/images/learning-guide/tutorials/rendering/vegetation-bending-tutorial/tree3.mp4" autoplay="true" loop="true" width="100%" muted="true" info="Three trees in the Editor, all swaying at a different pace with detail bending." >}}
+
+### Add motion vectors
+Because your tree is moving, you should add a *motion vector pass* so you can choose to add motion blur or other effects. For example, we have a [Temporal Anti-Aliasing (TAA)](/docs/atom-guide/features/taa.md) feature which uses the motion vectors.
+
+If you take a look at the `MeshMotionVectorVegetationBending.azsl` file, you can see that the vertex shader looks similar to the other vertex shaders you have seen, but there's a new output `OUT.m_worldPosPrev`. This is the position of the vector in the last time frame. The pixel shader uses both the previous vector position and the current one to calculate the motion vector. 
+
+However, there isn't a vertex shader input that gives us the previous position. Therefore, in the vertex shader, you will calculate the bending for not only the current time as you have been, but also for the previous time frame.
+
+Let's add the motion vector shader and then edit its vertex shader:
+
+1. Open `VegetationBending.materialtype`. At the bottom of the `shaders` list, add the motion vector pass:
+   
+   ```json
+   {
+      "file": "./MeshMotionVectorVegetationBending.shader",
+      "tag": "MeshMotionVector"
+   }
+   ```
+
+1. Open `MeshMotionVectorVegetationBending.azsl`.
+
+1. Add the call to `ProcessBending`, just as you did in the previous steps with the other shaders. Under the declaration for `float4 prevWorldPosition`, but above `OUT.m_worldPos`, add:
+
+   ```glsl
+   float currentTime = SceneSrg::m_time;
+   worldPosition = ProcessBending(currentTime, IN.m_position, IN.m_normal, IN.m_optional_color, worldPosition, objectToWorld);
+   ```
+
+1. For motion vectors to work you need to find the previous world position. You will want to perform bending on the vertex at the `IN.m_position` but at the previous frame time, with the previous world position. You can use `SceneSrg::m_prevTime` to get the previous frame time. Call your `ProcessBending` function again, but with the appropriate time and world position:
+
+   ```glsl
+   float currentTime = SceneSrg::m_time;
+   worldPosition = ProcessBending(currentTime, IN.m_position, IN.m_normal, IN.m_optional_color, worldPosition, objectToWorld);
+   float prevTime = SceneSrg::m_prevTime;
+   prevWorldPosition = ProcessBending(prevTime, IN.m_position, IN.m_normal, IN.m_optional_color, prevWorldPosition, prevObjectToWorld);
+   ```
```

Comment:
I suggest consolidating this step into a single step with substeps. I think it better wraps the idea of how motion vectors work. I also simplified the second step, regarding previous  frame times / world positions. This also let's you use one code block. 

Suggestions below results in this: 
![image](https://user-images.githubusercontent.com/75444793/182727427-1f327214-b255-47eb-b73a-2bb7ad746f43.png)

````suggestion
1. For motion vectors to work, you need to perform bending at the current frame time and the previous frame time.
 
   Under the `float4 prevWorldPosition` declaration, above `OUT.m_worldPos`, add the following:

   1. Call the `ProcessBending` function and pass in the current time and world position. This is similar to the calls you made in the earlier shaders.
   
   1. Call `ProcessBending` again, but this time, pass in the previous frame time and previous world position. Use `SceneSrg::m_prevTime` to get the previous frame time.

   ```glsl
   float currentTime = SceneSrg::m_time;
   worldPosition = ProcessBending(currentTime, IN.m_position, IN.m_normal, IN.m_optional_color, worldPosition, objectToWorld);
   
   float prevTime = SceneSrg::m_prevTime;
   prevWorldPosition = ProcessBending(prevTime, IN.m_position, IN.m_normal, IN.m_optional_color, prevWorldPosition, prevObjectToWorld);
   ```
````